www/api/class/centreon_metric.class.php

```diff
@@ -214,6 +252,286 @@ public function getMetricsDataByMetric()
+        ) {
+            throw new RestBadRequestException("Bad parameters");
+        }
+
+        /*
+         * Format:
+         *    {
+         *      "hostId_serviceId": {}, # no metricsId means: all metrics
+         *      "hostId2_serviceId2: { "metricId": 1, "metricId2": 1 }
+         *    }
+         */
+        $selectedMetrics = array();
+        if (is_string($services) && strlen($services) !== 0) {
+            foreach (explode(',', $services) as $service) {
+                list($hostId, $serviceId) = explode('_', $service);
+                if (false === is_numeric($hostId) ||
+                    false === is_numeric($serviceId)
+                ) {
+                    continue;
+                }
+                if (isset($selectedMetrics[$service])) {
+                    continue;
+                }
+
+                if ($this->checkAcl($hostId, $serviceId, $aclGroups, $isAdmin) <= 0) {
+                    continue;
+                }
+                $selectedMetrics[$service] = array();
+            }
+        }
+
+        if (is_string($metrics) && strlen($metrics) !== 0) {
+            $filter = '';
+            $filter_append = '';
+            foreach (explode(',', $metrics) as $metricId) {
+                if (false === is_numeric($metricId)) {
+                    continue;
+                }
+                $filter .= $filter_append . $metricId;
+                $filter_append = ',';
+            }
+
+            if ($filter === '') {
+                return $selectedMetrics;
+            }
+
+            $results = $this->pearDBMonitoring->query(
+                'SELECT 
+                    metric_id, host_id, service_id
+                 FROM metrics, index_data
+                 WHERE metrics.metric_id IN (' . $filter . ') AND metrics.index_id = index_data.id'
+            );
+
+            foreach ($results as $row) {
+                if (isset($selectedMetrics[$row['host_id'] . '_' . $row['service_id']]) && count($selectedMetrics[$row['host_id'] . '_' . $row['service_id']]) <= 0) {
+                    continue;
+                }
+                if (!isset($selectedMetrics[$row['host_id'] . '_' . $row['service_id']])) {
+                    if ($this->checkAcl($row['host_id'], $row['service_id'], $aclGroups, $isAdmin) <= 0) {
+                        continue;
+                    }
+                    $selectedMetrics[$row['host_id'] . '_' . $row['service_id']] = array();
+                }
+                $selectedMetrics[$row['host_id'] . '_' . $row['service_id']][$row['metric_id']] = 1;
+            }
+        }
+
+        return $selectedMetrics;
+    }
+
+    /**
+     * Get last data for metrics (by services and/or metrics)
+     *
+     * @param string $services List of services (like hostId_serviceId,hostId2_serviceId2,...)
+     * @param string $metrics  List of metrics (like metricId,metricId2,...)
+     *
+     * @return array
+     *
+     * @throws Exception
```

Comment:
Is this comment turning unchecked exception to checked exception ?
If not, describe the case that triggers it, or remove the comment.